src/main/scala/latis/service/hapi/DataService.scala

```diff
@@ -47,6 +47,11 @@ class DataService[F[_]: Concurrent](
           val records = alg.streamBinary(data)
           val stream: Stream[F, Byte] = if(req.header) Stream.emits(header.getBytes("UTF-8")) ++ records else records
           EitherT.right[Status](Ok(stream).map(_.withContentType(`Content-Type`(MediaType.application.`octet-stream`))))
+        case Json =>
+          val jsonHeader = io.circe.parser.parse(header.replaceAll("#", "")).toOption.get.asObject.get
```

Comment:
`.toOption.get.asObject.get` feels a little squirrelly, can you explain it please?

Reply:
I think rather than trying to parse the header it would be better to rearrange things a bit so the parsing wasn't necessary. So `header` would be `Json` or `InfoResponse` and turning it into a string for CSV/binary would be handled elsewhere. This would avoid the whole `.toOption.get.asOption.get` stuff.

Reply:
> `.toOption.get.asObject.get` feels a little squirrelly, can you explain it please?
Ya, I didn't totally like it but it got the types to work as I needed a JsonObject instead of Json to make adding `format:json` to the header easy. I'm open to trying to find a better way to do it.

Reply:
> I think rather than trying to parse the header it would be better to rearrange things a bit so the parsing wasn't necessary. So `header` would be `Json` or `InfoResponse` and turning it into a string for CSV/binary would be handled elsewhere. This would avoid the whole `.toOption.get.asOption.get` stuff.

That makes sense, I'll try that.